### ISSUE TITLE
about the implementation of sampling z

### ISSUE THREAD

**Issue by ShellingFord221** [JKDomoguen/Uncertainty-Aware-Pytorch-Implem#1]
Hi, in the paper, they say that

> we first sample random weights with dropout masks \omega and sample z such that z=g(x, \epsilon, \omega), \epsilon~N(0, I), with a pathwise derivative function g for reparameterization trick
(lines below Eq. 6)

and

> we first sample dropout masks \omega (s) ∼ q_M(\omega|X, Y) and then sample z~p_θ(z|x^*, \omega)
(line below Eq. 7)

I have two questions:
1. Why do they ignore KL divergence of q(z) and p(z) in loss function (i.e. Eq. 6)? Why are they equivalent? 
2. How do you implement 'first sample \omega, then sample z'? (I see that you sample z according to mu and sigma, but I don't what do they mean..)
Thanks!
